test/Microsoft.AspNetCore.Server.KestrelTests/EngineTests.cs

```diff
@@ -878,11 +878,7 @@ public void ConnectionCanReadAndWrite(ListenOptions listenOptions)
                     throw onStartingException;
                 }, null);
 
-                response.Headers["Content-Length"] = new[] { "11" };
-
-                var writeException = await Assert.ThrowsAsync<ObjectDisposedException>(async () =>
-                    await response.Body.WriteAsync(Encoding.ASCII.GetBytes("Hello World"), 0, 11));
-
+                var writeException = await Assert.ThrowsAsync<ObjectDisposedException>(async () => await response.Body.FlushAsync());
```

Comment:
Had to change so the check at https://github.com/aspnet/KestrelHttpServer/blob/e55c62444bdd34274173b011720c03f23c4db2f5/src/Microsoft.AspNetCore.Server.Kestrel/Internal/Http/FrameOfT.cs#L110 wouldn't close the connection (because now the write doesn't go through before `OnStarting` as it did before, that method sees 0 bytes written and that's a mismatch against the old test which wrote 11 bytes).